service/arm_templates/main.json

```diff
@@ -0,0 +1,125 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "clusterID": {
+      "type": "string",
+      "metadata": {
+        "description": "Unique ID of this guest cluster."
+      }
+    },
+    "storageAccountType": {
+      "type": "string",
+      "allowedValues": [
+        "Standard_LRS",
+        "Standard_GRS",
+        "Standard_ZRS",
+        "Premium_LRS"
+      ]
+    },
+    "virtualNetworkCidr": {
+      "type": "string",
+      "metadata": {
+        "description": "The main CIDR block reserved for this virtual network."
+      }
+    },
+    "masterSubnetCidr": {
+      "type": "string"
+    },
+    "workerSubnetCidr": {
+      "type": "string"
+    },
+    "apiLoadBalancerCidr": {
+      "type": "string"
+    },
+    "etcdLoadBalancerCidr": {
+      "type": "string"
+    },
+    "ingressLoadBalancerCidr": {
+      "type": "string"
+    },
+    "kubernetesAPISecurePort": {
+      "type": "string"
+    },
+    "etcdPort": {
+      "type": "string"
+    },
+    "kubernetesIngressSecurePort": {
+      "type": "string"
+    },
+    "kubernetesIngressInsecurePort": {
+      "type": "string"
+    },
+    "clusterSetupTemplate": {
```

Comment:
Could we have a single base URI parameter for all the templates? If you can use the `concat` function to build the template URIs that will reduce the number of params.

Reply:
Yes, I'll update the PR in a couple of minute to reflect that.